docs/DeveloperGuide.md

```diff
@@ -189,23 +189,37 @@ The `AddFavCommand#executeCommand()` method of AddFavCommand Class executes the
     - If the there are no duplicate Fav objects, Fav object created will be added to the FavList.
 
 ### 3.5. Favourite command executor (`/execfav` Feature)
-`/execfav <index>` is a command to execute a command with the specific index in the list of fovourite commands. <br><br>
+`/execfav <index>` is the command to execute a command with the specific index in the list of favourite commands. <br>
+
 The command is executed in the following steps:
-1. The user calls `Parser#setUserInput(<UserInput>)` by typing in the command `/execfav <index>`. The new user input is updated.
-2. `Parser#extractType()` is called to instantiate `FavList#getFav(<index>)` and run the user command.
+1. The user calls `Parser#setUserInput(<UserInput>)` by entering the command `/execfav <index>`. The new user input is updated.
+2. `Parser#extractType()` is called to instantiate `ExecFavCommand` and run the user command.
 3. `ExecFavParser` is instantiated and `ExecFavParser#setIndex()` is called to parse the `<index>` input from the user.
     - `ExecFavParser#setIndex()` method throws an exception if `<index>` cannot be parsed into an integer or if it is blank.
 4. `ExecFavCommand#executeCommand()` is called.
-5. `ExecFavCommand#getFav()` is self invoked to obtain the required Fav object from Favlist.
-    - An exception is thrown if there is no Fav object in the specified index in FavList.
-6. A new `Parser` object is instantiated to run the command in the Fav object.
-<br><br>
+5. `ExecFavCommand#getFav()` is self invoked to obtain the required `Fav` object from `FavList`.
+    - An exception is thrown if there is no `Fav` object in the specified index in `FavList`.
+6. A new `Parser` object is instantiated to run the command in the `Fav` object.
+    - If the command from the Fav object throws an exception when running, the exception will be caught in `ExecFavCommand#executeCommand()` and the `Fav` object will be deleted from the `FavList`.
 
 The following sequence diagram illustrates the steps taken by the program when the user calls the `/execfav` command.
 ![ExecFav_Sequence_Diagram](DG_Diagrams/ExecFavCommand/ExecFavCommand.png)
 
 ####Design Considerations
+##### Aspect: Choice of command object in FavList to execute
```

Comment:
You can talk about how typing full description is not very user friendly compared to just typing out numbers. Especially how some descriptions can be longer than the command itself.

Reply:
good idea!